app/src/main/java/com/example/studentarena/MainActivity.java

```diff
@@ -17,38 +25,72 @@
 import com.example.studentarena.fragments.MessageFragment;
 import com.example.studentarena.fragments.ProfileFragment;
 import com.google.android.material.bottomnavigation.BottomNavigationView;
+import com.parse.ParseGeoPoint;
 
 public class MainActivity extends AppCompatActivity {
+    private static final String TAG = "MainActivity";
+    private ParseGeoPoint userLocationGeoPoint;
+    private final FeedFragment fragmentHome = new FeedFragment(MainActivity.this);
+    private final ComposeFragment fragmentCompose = new ComposeFragment(MainActivity.this);
+    private final MessageFragment fragmentMessage = new MessageFragment();
+    private final ProfileFragment fragmentProfile = new ProfileFragment(MainActivity.this);
+    public BottomNavigationView bottomNavigationView;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
-        BottomNavigationView bottomNavigationView = findViewById(R.id.bottom_navigation);
+        bottomNavigationView = findViewById(R.id.bottom_navigation);
         bottomNavigationView.setOnItemSelectedListener(new BottomNavigationView.OnNavigationItemSelectedListener() {
             @Override
             public boolean onNavigationItemSelected(@NonNull MenuItem item) {
-                Fragment fragmentToShow = null;
+                Fragment fragmentToShow;
                 switch (item.getItemId()) {
                     case R.id.action_home:
-                        fragmentToShow = new FeedFragment();
+                        fragmentToShow = fragmentHome;
                         break;
                     case R.id.action_compose:
-                        fragmentToShow = new ComposeFragment();
+                        fragmentToShow = fragmentCompose;
                         break;
                     case R.id.action_message:
-                        fragmentToShow = new MessageFragment();
+                        fragmentToShow = fragmentMessage;
                         break;
                     case R.id.action_profile:
-                        fragmentToShow = new ProfileFragment();
+                        fragmentToShow = fragmentProfile;
                         break;
-                    default: break;
+                    default: fragmentToShow = fragmentHome;
                 }
-                if (fragmentToShow != null) {
                     getSupportFragmentManager().beginTransaction().replace(R.id.flContainer, fragmentToShow).commit();
-                }
                 return true;
             }
         });
+        setUserLocation();
         bottomNavigationView.setSelectedItemId(R.id.action_home);
     }
+    private void setUserLocation() {
+        LocationManager mLocationManager = (LocationManager) getSystemService(LOCATION_SERVICE);
+        long UPDATE_INTERVAL = 10000;  /* 10 secs */
+        long MIN_DISTANCE = 1 * 1609; /* 1 mile */
```

Comment:
google tells me 1 mile is 1609.34 meters but i had no idea what that 1609 was. maybe clarify that in the comment?